huaweicloud/services/workspace/resource_huaweicloud_workspace_service.go

```diff
@@ -654,7 +654,7 @@ func resourceServiceUpdate(ctx context.Context, d *schema.ResourceData, meta int
 		return diag.Errorf("error creating Workspace v2 client: %s", err)
 	}
 
-	if d.HasChanges("ad_domains", "access_mode", "dedicated_subnets") {
+	if d.HasChanges("ad_domain", "access_mode", "dedicated_subnets") {
```

Comment:
It seems that the test case does not cover this part of the change code.
It is recommended to add test cases to cover this part of the logic.